src/components/Login/Login.jsx

```diff
@@ -0,0 +1,252 @@
+import React, { useEffect, useState } from "react";
+
+import "./Login.css";
+import wavingPikachu from "../../img/waving-pikachu.png";
+import thorEndgame from "../../img/thor-endgame.png";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faUser } from "@fortawesome/free-solid-svg-icons";
+import { faLock } from "@fortawesome/free-solid-svg-icons";
+import { faFacebookF } from "@fortawesome/free-brands-svg-icons";
+import { faGoogle } from "@fortawesome/free-brands-svg-icons";
+import { faEnvelope } from "@fortawesome/free-solid-svg-icons";
+import { auth, gProvider, fProvider } from "../../fire";
+import { useDataLayerValue } from "../../context/DataLayer";
+
+function Login() {
+  const [{ user }, dispatch] = useDataLayerValue();
+  const [elementsObj, setElementsObj] = useState({});
+  const [userObj, setUserObj] = useState({});
+
+  const addSignUpMode = () => {
+    elementsObj.container.classList.add("sign-up-mode");
+  };
+
+  const remSignUpMode = () => {
+    elementsObj.container.classList.remove("sign-up-mode");
+  };
+
+  const handleSignUp = (e) => {
+    e.preventDefault();
+    auth
+      .createUserWithEmailAndPassword(userObj.email, userObj.password)
+      .then((user) => {
+        // signed up
+        var u = auth.currentUser;
+        u.updateProfile({
+          displayName: userObj.displayName,
+        })
+          .then(() => {
+            console.log("Set displayName");
+            dispatch({
+              type: "SET_USER",
+              user: u.providerData[0],
+            });
+          })
+          .catch((err) => {
+            console.log({ err });
+          });
+      })
+      .catch((error) => {
+        var errorCode = error.code;
+        var errorMessage = error.message;
+        console.log({ errorCode, errorMessage });
+      });
+  };
+
+  const handleSignIn = (e) => {
+    e.preventDefault();
+    auth
+      .signInWithEmailAndPassword(userObj.email, userObj.password)
+      .then((user) => {
+        console.log("user signed in", { user });
+      })
+      .catch((error) => {
+        var errorCode = error.code;
+        var errorMessage = error.message;
+        console.log({ errorCode, errorMessage });
+      });
+  };
+
+  const handleAuthLogin = (e) => {
+    e.preventDefault();
+    console.log(e.currentTarget.value);
+    switch (e.currentTarget.value) {
+      case "google":
+        auth.signInWithPopup(gProvider).catch((err) => alert(err.message));
+        break;
+      case "facebook":
+        auth.signInWithPopup(fProvider).catch((err) => alert(err.message));
```

Comment:
destructure err.message so we dont have 2 object look ups when we can replace with a variable